### DISCUSSION TITLE
Introducing our new feature voting portal!

### DISCUSSION THREAD

**Discussion by btueffers** [microsoft/WindowsAppSDK#1778] [Announcements]
In the Windows App SDK, being feedback driven is one of our primary goals, and we needed to find tooling that allowed us to efficiently collect and organize developer feedback from various sources (emails, meetings, online posts) into a single source of truth, so that we can focus on building what’s most important to you, and that we can keep everyone informed when features they asked for are delivered. 

We’ve been internally using [productboard](https://productboard.com) for several months now to aggregate customer feedback. We’re excited to announce that we’re opening our productboard [portal](https://portal.productboard.com/winappsdk/1-windows-app-sdk) to the public as the new location to submit and vote on feature requests. The portal will also serve as our roadmap where we will document when features are expected to arrive. 

**Where should I vote on or suggest features going forward?**

Use our [public portal](https://portal.productboard.com/winappsdk/1-windows-app-sdk)! If you don’t see a feature, you can suggest a new feature in the top right. Those suggestions are private and only become visible to the public when the Microsoft team wants to gauge community interest in the feature. 

[![image](https://user-images.githubusercontent.com/82911735/141384693-3c588225-2022-4ad7-bb9b-3247ffe26692.png)](https://portal.productboard.com/winappsdk/1-windows-app-sdk)

**We’re still using GitHub!**

GitHub is still our go-to resource for issues (bugs), discussions, specs, development, and more! We are using productboard for a collaborative community roadmap to supplement team resources. Microsoft’s Adaptive Cards team has seen a dramatically positive impact on their feedback process using [their portal](https://portal.productboard.com/adaptivecards/1-adaptive-cards-features) compared to exclusively GitHub with both better quantity and quality. 

**Key reasons we’re using productboard…**

- Feedback can be consolidated from multiple sources like meeting notes, direct customer comments, and... GitHub! Which will lead to fewer overlooked opinions.  
- Allows the team to more easily notify developers when a feature they're interested in has shipped or is available in a preview release. 
- A productboard account isn’t required for community members to vote on features, which removes a barrier to participation. 
- Feedback is private so developers can more confidently share details about their requests! 
- Improved requirements gathering through productboard features such as requiring more targeted answers. For example, must have, should have, nice to have. 
- Previous success with the [Adaptive Cards portal](https://portal.productboard.com/adaptivecards/1-adaptive-cards-features)! At times we’ve seen up to 4x more feedback from the community along with more thorough explanations. All leading to stronger customer relationships!

**Comment by wbokkers** [comment]
This is great! Now implement this multiple-window support in WinUI 3 please!!

**Comment by Panda-Sharp** [comment]
Why did you decide, to come up with another portal for feedbacks? most of MS products finally are aligning to use GitHub or feedbackportal.microsoft.com (instead of uservoice) ... this is so MS that a team decide to use and add a new tool to the list once again, adding confusion instead of having everything in one place

**Reply by andrewleader** [reply]
Hi @Panda-Sharp, great question. We touched on "Why not GitHub" here, we've seen great success when our Adaptive Cards product (that started on GitHub) started using productboard for feature requests... the number of votes and feedback customers started providing started to skyrocket compared to GitHub. In addition, productboard provides better planning tooling so we can take that feedback and keep that feedback at the forefront of all of our planning, ensuring feedback doesn't get lost or forgotten.

As for why not the new Feedback Portal, we investigated using that too, and would love to use that someday, however it doesn't meet our smaller product needs today. We get a lot of feedback from meetings with customers and emails from customers that we need to be able to aggregate into a single source of feedback, and the Feedback Portal currently doesn't help with that. We've used UserVoice previously in the Windows developer platform, and for several reasons it didn't prove to be effective for a smaller product like ours. But for larger product teams (which it was designed for), I think Feedback Portal works really well.

**Reply by Panda-Sharp** [reply]
Hi @andrewleader thanks for your reply, correct me if I'm wrong, but the main issues that I see with productboard are for example:
- I cannot login with my MS account so I cannot see a list of feedbacks I already submitted and voted
- When I click on submit  a new idea, there is no way for a feedback already submitted (from me or someone else), to avoid duplicates and double work for you
I think you already considered this, and I'm curios to know your reasons why you still decided for productboard
thanks

**Reply by andrewleader** [reply]
If you vote on an item again, once you confirm your email address again, you'll be "signed in" and be able to see the items you've already voted on. But yeah there's no way to just sign in from the start to see your previous items, I'll share that feedback with them.

As for duplicate new ideas, it's super easy for us to link new ideas to existing items. Additionally, sometimes the "same" idea from two different people actually have subtle differences, or they might use different terminology and not even realize they're talking about the same thing, so since we manually link those, we can be more sure the feedback is being linked up correctly.  The new idea submissions have been easily manageable by our team so far.

**Comment by wbokkers** [comment]
I really hope that the portal is not yet another feedback mechanism where feedback largely gets ignored. 
And what I really don't like about the portal is that it is encouraging feature creep. Major issues already get ignored in GitHub. Now there's another distraction in place for Microsoft employees to ignore those issues.

**Reply by andrewleader** [reply]
Hey @wbokkers, thanks for the feedback, 100% agreed with your concerns. Avoiding a "black hole" of feedback is explicitly the reason we chose productboard. We do our planning out of productboard, we track our partner feature requests through productboard, etc. Productboard allows us to aggregate all of our feedback (meetings from customers, emails from customers, public votes from the community, etc) into a single source, so that we don't have that feedback spread across a lot of places that are subject to being ignored.

![image](https://user-images.githubusercontent.com/13246069/142463630-549ff7ee-d7c9-4454-8911-5c92a19ec021.png)

Since it's integrated with our planning tools, our hope is that this will ensure customer feedback stays at the forefront of every decision and every moment. When we're discussing what features to build, we can easily ask "which customers asked for this?" and easily see that in productboard. Previously, we would have to try to manually aggregate that info across GitHub and recollection of meetings with customers.

**Reply by andrewleader** [reply]
And by the way, if you're expressing fears of feedback getting ignored because media controls currently isn't in the 1.1 tentative plans, I explained that more here: https://github.com/microsoft/WindowsAppSDK/discussions/817#discussioncomment-1666081

**Reply by wbokkers** [reply]
@andrewleader My fear is that serious issues don't get resolved, like: https://github.com/microsoft/microsoft-ui-xaml/issues/6218 and https://github.com/microsoft/microsoft-ui-xaml/issues/4574. There's a tendency of ignoring WinRT/XAML related issues.

**Comment by wjk** [comment]
I notice that [following our 1.1 release](https://github.com/microsoft/WindowsAppSDK/discussions/2567), all of the items on the Planned tab have been completed. However, ProductBoard has not been updated in a long time. What are the plans for what is next with App SDK/WinUI 3? We have a chunk of ideas on the Backlog that were agreed upon long, long ago but always punted “to the next release.” Can we give them some love?

Furthermore, I have heard rumors that now that WinUI 3 has been fully released, that it will be put into maintenance mode and the team will be reassigned to chase the next shiny-shiny. This would be a **terrible** idea, especially considering that WinUI 3 is (still!) not open source. Complaints from the community that Microsoft keeps killing and recreating developer frameworks are caused by exactly this behavior. (Windows Forms and WPF are still the most widely compatible .NET UI toolkits in existence, partially because they fully accept community contributions and there is no interference from the Windows “business model says no” guys.) You promised that App SDK would be fully detached from the OS, but this is unfortunately not true, since many of its newer features require Windows 11. The fact that you _still_ cannot install Windows 11 on existing hardware, even in a virtual machine (and I’ve tried, twice!) renders much of WinUI 3 irrelevant. Thoughts?

**Reply by Gavin-Williams** [reply]
> You promised that App SDK would be fully detached from the OS

It's WindowsAppSDK, not AppSDK. I think you misunderstood.

**Reply by Balkoth** [reply]
I think you missed his point. The initial promise was that all functionality would be available from a set windows 10 version. As this started to grow, more and more features crept in that downright crashed the app on windows 10. Bold claims were made and outright failed to deliver.

**Comment by meirumeiru** [comment]
I don't know if I like this "feedback driven development". There was a time when "test driven development" was "the thing". Everyone started using it and it was seen as "the only truth". Now... I think, many of us agree on the fact, that this has been proven wrong. And today, I have the same feeling for "feedback driven development". Why? Well, I can tell you why. Because the things that are wanted most by people are not always the things that they need most. They talk a lot about the "cool" features. And the request stuff like shaking buttons and blinking edit windows and whatever. But that's not the important stuff that we need. We need a solid foundation to build on. And only about 1 out of 10'000 developers have the knowledge to talk about those kind of things. ... same thing is true for "telemetry driven development". If Microsoft starts deprecating APIs that are "not often used"... what would the result be? That they're deprecating APIs used in the SQL server to read/write data? Just because there are "only" 10 applications worldwide using it but millions of stupid hobby "apps" that don't? ... so the ration is 1/100'000... then... let's drop those APIs? ... as I said... I think it's a bad idea and it binds too many human resources. Sure, feedback is good... but not every feedback. Most of it is simply stupid.
